src/ExtractTileOperations.cpp

```diff
@@ -39,12 +39,33 @@ Type amx_op_type_result_type(AMXOpType op_ty) {
     }
 }
 
+int amx_op_type_size(AMXOpType op_ty) {
+    switch (op_ty) {
+    case AMXOpType::Int8:
+        return 1;
+    case AMXOpType::Bfloat16:
+        return 2;
+    default:
+        internal_error << "Unexpected";
+        return -1;
+    }
+}
+
 const auto wild_i32 = Variable::make(Int(32), "*");
 const auto wild_i32x = Variable::make(Int(32, 0), "*");
 
 Tile<1> get_1d_tile_index(const Expr &e) {
     if (const auto *r1 = e.as<Ramp>()) {
-        return {true, r1->base, {r1->stride}, {r1->lanes}};
+        Expr pattern = ((wild_i32 * wild_i32) + wild_i32) * wild_i32;
+
+        std::vector<Expr> results;
+        if (!expr_match(pattern, r1->base, results)) {
+            return {};
+        }
+
+        auto stride = std::move(results[1]);
```

Comment:
Why is the stride the second arg to the multiplication? It could easily be the first, no?

Reply:
I have added a check for the stride being an `IntImm`, that should prevent the other term being mistakenly matched.